.cirrus.yml

```diff
@@ -0,0 +1,40 @@
+task:
+  env:
+    HOME: /tmp  # cargo cache needs it
+    VERSION: nightly
+  matrix:
+    - name: FreeBSD 12 amd64 nightly   
+      freebsd_instance:
+        image: freebsd-12-3-release-amd64
+    - name: FreeBSD 13 amd64 nightly
+      freebsd_instance:
+        image: freebsd-13-1-release-amd64
+    - name: FreeBSD 14 amd64 nightly
+      freebsd_instance:
+        image: freebsd-14-0-snap
+    - name: FreeBSD 13 amd64 stable
+      env:
+        VERSION: 1.62.0
+      freebsd_instance:
+        image: freebsd-13-1-release-amd64
+  setup_script:
+    - fetch https://sh.rustup.rs -o rustup.sh
+    - sh rustup.sh -y --profile=minimal
+    - $HOME/.cargo/bin/rustup toolchain install $VERSION
+  cargo_cache:
+    folder: $HOME/.cargo/registry
+    fingerprint_script: cat Cargo.lock || echo ""
+  test_script:
+    - . $HOME/.cargo/env
+    - cargo +$VERSION test
+  lint_script:
```

Comment:
It would be nice to have a `rustfmt` task as well, but we can do this in a follow up since it requires a `cargo fmt` run.